File: Lab5/waa_lab_5/src/main/java/com/example/waa_lab_5/entity/Post.java
package com.example.waa_lab_5.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Setter
@Getter
public class Post {

    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true)
    @JoinColumn(name = "post_id")
    List<Comment> comments;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;
    private String title;
    private String content;
    private String author;
}

File: Lab7/waa-lab-7/src/components/Post/Post.js
import React from "react";
import "./Post.css";

const Post = (props) => {
  return (
    <div className="post" onClick={props.onClick}>
      <h2>Id : {props.id}</h2>
      <h2>Title : {props.title}</h2>
      <p>Content : {props.content}</p>
      <p>Author : {props.author}</p>
    </div>
  );
};

export default Post;
